src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/PreferStreamAsyncMemoryOverloads.Fixer.cs

```diff
@@ -0,0 +1,214 @@
+
+            // Generate the new buffer argument, ensuring we include the argument name if the user originally indicated one
+            SyntaxNode namedBufferNode = isBufferNamed ? generator.Argument("buffer", RefKind.None, asMemoryInvocationNode) : asMemoryInvocationNode;
+
+            // Create an async method call for the stream object with no arguments
+            SyntaxNode asyncMethodNode = generator.MemberAccessExpression(streamInstanceNode, methodName);
+
+            // Add the arguments to the async method call, with or without CancellationToken
+            SyntaxNode[] nodeArguments;
+            if (cancellationTokenValueNode != null)
+            {
+                SyntaxNode namedCancellationTokenNode = isCancellationTokenNamed ? generator.Argument("cancellationToken", RefKind.None, cancellationTokenValueNode) : cancellationTokenValueNode;
+                nodeArguments = new SyntaxNode[] { namedBufferNode, namedCancellationTokenNode };
+            }
+            else
+            {
+                nodeArguments = new SyntaxNode[] { namedBufferNode };
+            }
+            SyntaxNode newInvocationExpression = generator.InvocationExpression(asyncMethodNode, nodeArguments);
+
+            SyntaxNode newRoot = generator.ReplaceNode(root, invocation.Syntax, newInvocationExpression);
+            return Task.FromResult(doc.WithSyntaxRoot(newRoot));
+        }
+    }
+
+    [ExportCodeFixProvider(LanguageNames.CSharp)]
+    public sealed class PreferStreamAsyncMemoryOverloadsCSharpFixer : PreferStreamAsyncMemoryOverloadsFixer
+    {
+        protected override IArgumentOperation? GetArgumentByPositionOrName(ImmutableArray<IArgumentOperation> args, int index, string name, out bool isNamed)
+        {
+            isNamed = false;
+
+            // The expected position is beyond the total arguments, so we don't expect to find the argument in the array
+            if (index >= args.Length)
+            {
+                return null;
+            }
+            // If the argument in the specified index does not have a name, then it is in its expected position
+            else if (args[index].Syntax is CodeAnalysis.CSharp.Syntax.ArgumentSyntax argNode && argNode.NameColon == null)
+            {
+                return args[index];
+            }
+            // Otherwise, find it by name
+            else
+            {
+                isNamed = true;
+                return args.FirstOrDefault(argOperation =>
+                {
+                    return argOperation.Syntax is CodeAnalysis.CSharp.Syntax.ArgumentSyntax argNode &&
+                           argNode.NameColon != null &&
+                           argNode.NameColon.Name != null &&
+                           argNode.NameColon.Name.Identifier != null &&
+                           argNode.NameColon.Name.Identifier.ValueText == name;
```

Comment:
```suggestion
                           argNode.NameColon?.Name.Identifier.ValueText == name;
```

Reply:
Ah thanks. I always forget about this relatively new nullability feature.